### PR TITLE
Feature/gov 56 complete signup

### PR DESCRIPTION
**Pull request by aleciafb**
# Background

> As a Newly Added/ Invited User I want to be able to complete my registration/signup as a user on the GovChat management portal so that I can login and use the GovChat management portal and have access to the correct ward or municipality information.

Short version
* Admin invite user
* User receives email 
* Click link in the email and complete signup 

# What was done
**API**
- Verify token endpoint
- Password set endpoint *TODO*
- Patch user endpoint
- Still show errors when status 401 

**Actions**
- Verify token
- Update user
- Set password *TODO*
- Sign up user *TODO*

**Miscellaneous**
- Move alerts out of LandingLayout
- Fix duplicate id error on select wrapper

**Components and containers**
- Signup 
- Terms of Service
- Privacy Policy

**Route subscribers**
- Verify token
- Sign up

# Supporting Info
https://praekelt.atlassian.net/browse/GOV-56

# TODO
- [x] Password validation
- [x] Terms accepted validation
- [x] Update Terms of service accepted 
- [x] Set password enpoint 
- [x] Set password action 
- [x] Signup action
- [x] Test 204 on API layer 
- [x] Set invite pending flag when inviting user
- [x] Terms of Service Page
- [x] Privacy Policy page
- [x] Link to Terms of Service and Privacy policy
- [x] Fix checkbox
- [x] Handle checkbox errors
- [x] Styling
 

- [x] Validate ward and municipality based on role
- [ ] Verify token flow ***BLOCKED***
- [x] Update verify token route on API side
- [x] Remove console.logs
- [x] Remove initValues spread
- [x] Remove unused props
- [x] Remove unused imports

### CODE REVIEW

File: src/actions/userTypes.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ export const fetchCurrentUserType = () => async (dispatch, getState) => {
     await dispatch(fetchUserById(currentUserId));
     const currentUser = selectors.getUserById(getState(), currentUserId);
 
+    console.log(currentUser);
```
Comment (aleciafb) [inline src/actions/userTypes.js]: remove

File: src/components/Signup/index.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            </Typography>
+            <Typography variant="subheading" color="inherit">
+                Complete your govChat account signup and registration by filling
+                in the form below
+            </Typography>
+        </div>
+        <SignupForm
+            onSubmit={handleSubmit}
+            classes={classes}
+            initialValues={{ ...initialValues }}
```
Comment (aleciafb) [inline src/components/Signup/index.js]: Remove spread operator

File: src/containers/Signup.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        user: user,
+        wardId: wardId,
+        // userType: getCurrentUserType(state)
+        userType: 'Ward Councillor'
+    };
+};
+
+const mapDispatchToProps = dispatch => ({
+    handleSubmit(formData) {
+        console.log('Handled');
```
Comment (aleciafb) [inline src/containers/Signup.js]: remove

File: src/routeSubscribers.js [hunk 4]
```diff
@@ -36,6 +41,8 @@ const routeSubscriber = ({ dispatch, getState, subscribe }) => {
         )
         .map(state => (state.router.location || {}).pathname);
 
+    console.log(loggedInPath$, path$);
```
Comment (aleciafb) [inline src/routeSubscribers.js]: remove

File: .env.development [hunk 5]
```diff
@@ -5,3 +5,6 @@ REACT_APP_INVITE_NOTIFICATION='System User Registration Invitation'
 REACT_APP_DEACTIVATION_NOTIFICATION='System User Deactivation'
 REACT_APP_ACTIVATION_NOTIFICATION='System User Reactivation'
 REACT_APP_ADMIN='Super Administrator'
+REACT_APP_MUNICIPAL_MANAGER='Municipal Manager'
```
Comment (aleciafb) [inline .env.development]: @jwfwessels Spoke to Jono, we will look at managing permissions differently on the FE. Ticket in the backlog.

File: src/actions/api/users.js [hunk 6]
```diff
@@ -77,6 +79,53 @@ export const createUser = ({
...
+            apiFetch(`${USERS_ENDPOINT}${id}/`, {
+                method: 'PATCH',
+                body: JSON.stringify({
+                    id: id,
+                    username: username,
+                    password: password,
+                    first_name: firstName,
+                    last_name: lastName,
+                    email: email,
+                    is_active: true,
```
Comment (aleciafb) [inline src/actions/api/users.js]: set to true on signup

File: src/components/App/index.js [hunk 7]
```diff
@@ -8,13 +8,16 @@ import UserInvite from '../../containers/UserInvite';
 import AlertMessages from '../../containers/AlertMessages';
 import Login from '../../containers/Login';
 import LandingLayout from '../../components/LandingLayout';
+import Signup from '../../containers/Signup';
+import TermsOfService from '../../components/TermsOfService';
```
Comment (aleciafb) [inline src/components/App/index.js]: These are placeholders for now - will pull actual content once we have everything in place

File: src/components/Signup/index.js [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+                color="primary"
+                className={classes.formGroupHeading}
+            >
+                Administrative Information
+            </Typography>
+            <Grid container spacing={24}>
+                <Grid item xs={6} sm={6}>
+                    {userType && (
+                        <div>
+                            {(userType ===
```
Comment (aleciafb) [inline src/components/Signup/index.js]: As mentioned before, these checks will be replaced with proper authorization matrix or something

File: src/containers/Signup.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    getUserById,
+    getWards,
+    getJurisdictions,
+    getCurrentUserType
+} from '../reducers';
+import * as userActions from '../actions/users';
+
+const mapStateToProps = state => {
+    //TODO Change to currentUserID - ONly debug purposes
+    const user = getUserById(state, 21);
```
Comment (aleciafb) [inline src/containers/Signup.js]: Will be changed to `currentUserID` and no hard coded role, once GOV-305 is resolved https://praekelt.atlassian.net/browse/GOV-305

File: src/routeSubscribers.js [hunk 10]
```diff
@@ -74,6 +79,25 @@ const routeSubscriber = ({ dispatch, getState, subscribe }) => {
             await dispatch(fetchWards());
             await dispatch(fetchJurisdictions());
         });
+
+    const userSignupPath = pathToRegexp('/userprofile/signup');
+    loggedInPath$
+        .filter(path => userSignupPath.test(path))
+        .subscribe(async () => {
+            const id = selectors.getCurrentUserId(getState());
+            // TODO Change to ID
```
Comment (aleciafb) [inline src/routeSubscribers.js]: same here

File: src/actions/api/users.js [hunk 11]
```diff
@@ -77,6 +79,54 @@ export const createUser = ({
...
+        : null;
+    const ward = wardId
+        ? `${API_ROOT}${API_VERSION}${WARDS_ENDPOINT}${wardId}/`
+        : null;
+
+    try {
+        return await dispatch(
+            apiFetch(`${USERS_ENDPOINT}${id}/`, {
+                method: 'PATCH',
+                body: JSON.stringify({
```
Comment (jwfwessels) [inline src/actions/api/users.js]: 2 things:
- arent you using normalizr?
- no need to specify pairs if the property name remains the same.

File: src/containers/Signup.js [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+    if (user) {
+        wardId = user.ward && Number(getResourceIdFromUrl(user.ward));
+        jurisdictionId =
+            user.jurisdiction &&
+            Number(getResourceIdFromUrl(user.jurisdiction));
+
+        initialValues = {
+            ...user,
+            cellPhone: user.cell_phone,
```
Comment (jwfwessels) [inline src/containers/Signup.js]: normalization again 🤔

File: src/actions/api/users.js (hunk 11 above)
Comment (aleciafb) [reply src/actions/api/users.js]: 👍

**Pull request merged**